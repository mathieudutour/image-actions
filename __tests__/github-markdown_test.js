const results = {
  optimisedImages: [
    {
      name: 'icon.png',
      path: '__tests__/test-images/icon.png',
      beforeStats: 8914,
      afterStats: 3361,
      percentChange: -62.29526587390622,
      compressionWasSignificant: true
    },
    {
      name: 'roo.jpg',
      path: '__tests__/test-images/roo.jpg',
      beforeStats: 485742,
      afterStats: 468895,
      percentChange: -3.592915258213452,
      compressionWasSignificant: true
    }
  ],
  unoptimisedImages: [
    {
      name: 'optimised-image.png',
      path: '__tests__/test-images/optimised-image.png',
      beforeStats: 3361,
      afterStats: 3361,
      percentChange: 0,
      compressionWasSignificant: false
    },
    {
      name: 'another-optimised-image.png',
      path: '__tests__/test-images/another-optimised-image.png',
      beforeStats: 3361,
      afterStats: 3361,
      percentChange: 0,
      compressionWasSignificant: false
    }
  ],
  metrics: {
    bytesSaved: 5553,
    percentChange: -62.29526587390622
  }
}

const markdown = require('../dist/github-markdown').default

test('writes the markdown report', async () => {
  const markdownResult = await markdown({
    processingResults: results
  })

  expect(markdownResult).toMatchSnapshot()
})
