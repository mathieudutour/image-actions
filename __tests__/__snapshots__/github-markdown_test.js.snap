// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writes the markdown report 1`] = `
"Images automagically compressed by âœ¨

Compression reduced images by <strong>62.3%</strong>, saving <strong>5.42 KB</strong>.

| Filename | Before | After | Improvement |
| -------- | ------ | ----- | ----------- |
| <code>icon.png</code> | 8.71 KB | 3.28 KB | -62.3% |
| <code>roo.jpg</code> | 474.36 KB | 457.91 KB | -3.6% |


2 images did not require optimisation.

"
`;
